mantidproject/mantid
# Add InstrumentView to ISIS SANS GUI

~~**Do not merge or test before #20229 has been merged**~~

This PR allows the user to see the moved and masked workspace in the InstrumentView.

Fixes #20283

**To test:**

Please find test data here: smb://olympic/babylon5/Scratch/Anton/GUI_TEST

1. Open the ISIS SANS v2 experimental interface
1. Select `SANS2D` as the instrument (below the data table)
2. For the user file load: USER_SANS2D_154E_2p4_4m_M3_Xpress_8mm_SampleChanger_FRONT.txt
3. For the batch file load: batch_file.csv
4. Go the `Settings>Mask` tab
5. Press `Display Mask`
   * Confirm that the GUI is still responsive
   * Confirm that the button is disabled and has changed its label
   * Confirm that after a while the InstrumentView with the masked data is displayed
   * Confirm that the button is enabled again and has the correct label



**Release Notes** 
Not needed

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
